File: src/main/kotlin/chapter6/java/Journey.java
package chapter6.java;

public class Journey {

    private String name;

    public Journey(String name) {
        this.name = name;
    }

    public Location getDepartsFrom() {
        return new Location("location");
    }

    public static <U extends Comparable<? super U>, T> int getDuration(T t) {
        return 0;
    }
}

File: src/main/kotlin/chapter6/kotlin/Journey.kt
package chapter6.kotlin

data class Journey(
    var name: String,
    var duration: Int
) {
    fun getDepartsFrom(): Location {
        return Location("location")
    }

    fun <U : Comparable<U?>?, T> getDuration(t: T): Int {
        return 0
    }
}